instrumentation/action_view/test/opentelemetry/instrumentation/action_view/span_subscriber_test.rb

```diff
@@ -157,4 +157,40 @@
       _(last_span.attributes['thing']).must_equal('optimus prime')
     end
   end
+
+  describe 'instrument' do
+    before do
+      ::ActiveSupport::Notifications.unsubscribe('bar.foo')
+    end
+
+    it 'does not trace an event by default' do
+      ::ActiveSupport::Notifications.subscribe('bar.foo') do
+        # pass
+      end
+      ::ActiveSupport::Notifications.instrument('bar.foo', extra: 'context')
+      _(last_span).must_be_nil
+    end
+
+    it 'traces an event when a span subscriber is used' do
+      ::OpenTelemetry::Instrumentation::ActionView.subscribe('bar.foo', subscriber)
+      ::ActiveSupport::Notifications.instrument('bar.foo', extra: 'context')
+
+      _(last_span).wont_be_nil
+      _(last_span.name).must_equal('foo bar')
+      _(last_span.attributes['extra']).must_equal('context')
+    end
+
+    it 'finishes spans even when other subscribers blow up' do
+      ::OpenTelemetry::Instrumentation::ActionView.subscribe('bar.foo', subscriber)
+      ::ActiveSupport::Notifications.subscribe('bar.foo') { raise 'boom' }
```

Comment:
I think we should change the ordering to this, or have this as a separate test case as well.  It asserts that our forced ordering logic works.
```suggestion
      ::ActiveSupport::Notifications.subscribe('bar.foo') { raise 'boom' }
      ::OpenTelemetry::Instrumentation::ActionView.subscribe('bar.foo', subscriber)
```

In our span subscriber we define the `start` and `finish` methods so we can start and finish our span.

It would be nice to have a subscriber that raises on the start hook, and one that raises on the finish hook in addition to the existing tests I think it would a lot of confidence that this is the right solution.